pkg/graveler/graveler.go

```diff
@@ -1016,14 +1020,43 @@ func (g *Graveler) ListRepositories(ctx context.Context) (RepositoryIterator, er
 	return g.RefManager.ListRepositories(ctx)
 }
 
-func (g *Graveler) WriteRange(ctx context.Context, repository *RepositoryRecord, it ValueIterator) (*RangeInfo, error) {
+func (g *Graveler) WriteRange(ctx context.Context, repository *RepositoryRecord, it ValueIterator) (*RangeInfo, []ValueRecord, error) {
 	return g.CommittedManager.WriteRange(ctx, repository.StorageNamespace, it)
 }
 
 func (g *Graveler) WriteMetaRange(ctx context.Context, repository *RepositoryRecord, ranges []*RangeInfo) (*MetaRangeInfo, error) {
 	return g.CommittedManager.WriteMetaRange(ctx, repository.StorageNamespace, ranges)
 }
 
+func (g *Graveler) StageObjects(ctx context.Context, objects []ValueRecord, stagingToken string) (string, error) {
+	st := StagingToken(stagingToken)
+	if stagingToken == "" {
+		st = GenerateStagingToken("import", "ingest_range")
+	}
+	for _, obj := range objects {
+		err := g.StagingManager.Set(ctx, st, obj.Key, obj.Value, false)
```

Comment:
The graveler API requires a branchID which we do not have

Reply:
Yikes! Still - let's iterate from the outside and make this call handle a single key. It will keep the operation atomic at the graveler level when it fails half way through.